contracts/FeralfileArtworkV4_2.sol

```diff
@@ -0,0 +1,221 @@
+        } else {
+            if (saleData_.price != msg.value) {
+                revert InvalidPaymentAmount();
+            }
+        }
+
+        uint256 itemRevenue;
+        if (saleData_.price > saleData_.cost) {
+            itemRevenue =
+                (saleData_.price - saleData_.cost) /
+                saleData_.quantity;
+        }
+
+        uint256 nextPurchasableTokenId = seriesNextPurchasableTokenIds[saleData_.seriesID];
+        uint256 i = 0;
+        while (i < saleData_.quantity) {
+            uint256 tokenIdForSale = nextPurchasableTokenId;
+
+            if (!_exists(tokenIdForSale)) {
+                revert TokenIDNotFound();
+            }
+
+            nextPurchasableTokenId++;
+            if (ownerOf(tokenIdForSale) != address(this)) {
+                continue;
+            }
+
+            // send NFT
+            _safeTransfer(
+                address(this),
+                saleData_.destination,
+                tokenIdForSale,
+                ""
+            );
+
+            emit BuyArtworkV2(
+                saleData_.destination,
+                tokenIdForSale,
+                saleData_.nonce
+            );
+            i++;
+        }
+
+        // save next sale token id for seriesID
+        seriesNextPurchasableTokenIds[saleData_.seriesID] = nextPurchasableTokenId;
+
+        // distribute royalty
+        uint256 distributedRevenue;
+        uint256 platformRevenue;
+
+        RevenueShare[] memory revenueShares = saleData_.revenueShares;
+        uint256 remainingRev = itemRevenue;
+
+        // deduct advances payment from revenue
+        for (uint256 j = 0; j < revenueShares.length && remainingRev > 0; j++) {
+            uint256 remainingAdvanceAmount = advances[
+                revenueShares[j].recipient
+            ];
+            uint256 rev = remainingAdvanceAmount >= remainingRev
+                ? remainingRev
+                : remainingAdvanceAmount;
+            uint256 totalRev = saleData_.quantity * rev;
+            platformRevenue += totalRev;
+            advances[revenueShares[j].recipient] -= totalRev;
+            remainingRev -= totalRev;
+        }
+
+        // distribute revenue
+        if (remainingRev > 0) {
+            for (uint256 j = 0; j < revenueShares.length; j++) {
+                address recipient = revenueShares[j].recipient;
+                uint256 rev = (remainingRev * revenueShares[j].bps) / 10000;
+                uint256 totalRev = saleData_.quantity * rev;
+                if (recipient == costReceiver) {
+                    platformRevenue += totalRev;
+                    continue;
+                }
+                distributedRevenue += totalRev;
+                payable(recipient).transfer(totalRev);
+            }
+        }
+
+        if (
+            saleData_.price - saleData_.cost <
+            distributedRevenue + platformRevenue
+        ) {
+            revert TotalBpsOver();
+        }
```

Comment:
It's a miss from long time ago that we should add a check for `saleData_.price > saleData_.cost`

Reply:
or we could discuss should we allow to submit a buy tx that cost larger than price.